src/zck_gen_zdict.c

```diff
@@ -248,36 +289,47 @@ int main (int argc, char *argv[]) {
     snprintf(out_name + strlen(base_name) - 4, 7, ".zdict");
 
     if(!zck_close(zck)) {
-        dprintf(STDERR_FILENO, "%s", zck_get_error(zck));
+        LOG_ERROR("%s", zck_get_error(zck));
         goto error2;
     }
 
     /* Create dictionary */
+#ifdef _WIN32
+    char buf[5000];
+    sprintf(buf, "zstd --train '%s' -r -o '%s'", dir, out_name);
+    int w = system(buf);
```

Comment:
Does Windows not support `execl`?  I'm leery of sending this command as a string rather than an array.

Reply:
Nope, as far as I know execl doesn't exist on Windows (and hardcoded `/usr/bin/zstd` doesn't exist, either). 
I think `system` is the easiest way as it's supported in plain C. Otherwise we need to work with Win32 API, e.g. https://stackoverflow.com/questions/44105982/win32-c-api-alternative-to-broken-execl-family-of-functions ...

Also `fork()` doesn't really exist on Windows (only cygwin seems to have an implementation).

Reply:
Ok, that makes sense.

Reply:
https://nibblestew.blogspot.com/2019/11/process-invocation-will-forever-be.html

On Windows, all commands will be sent as strings, not as arrays. Windows helpfully provides the `_exec` (and `_spawn`, which is separately documented but seemingly identical?) family of functions, which like the POSIX `exec` family of functions accepts arguments as an array, and unlike the POSIX `exec` family of functions then converts it into a string by concatenating with spaces and calling CreateProcess on it.

In short, it doesn't matter what you do. You will never, ever, ever, ever be able to call processes sanely using arrays to safely convey arguments on Windows. Trying is a fool's errand. Thinking you have found an answer will lead to heartbreak when Windows says "yes I have a function that takes arrays" and then the Windows standard library converts your array to a string behind your back.

Since everything on Windows is equally as bad as `system()`, you might as well just use it with your basic attempts at quoting, which are better than the ones Windows provides.